cxx/transition_true_value.hh

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#pragma once
+
+#include <algorithm>
+#include <random>
+#include <string>
+#include <unordered_map>
+
+#include "emissions/base.hh"
+#include "noisy_relation.hh"
+#include "relation.hh"
+
+typedef int T_item;
+typedef std::vector<T_item> T_items;
+
+// Updates the estimate of a latent value conditioned on noisy observations.
+template <typename ValueType>
+void transition_true_value(
+    std::mt19937* prng, T_items base_items, Relation<ValueType>* base_relation,
+    std::unordered_map<std::string, NoisyRelation<ValueType>*>
+        noisy_relations) {
+  // Since this method does not transition cluster assignments, we incorporate
```

Comment:
Some of this comment might be better placed in the over-all function documentation, as it describes its interface and invariants.  (i.e., "This method does not transition cluster arguments.  Empty clusters are not deleted, and the number of observations in each cluster is the same before and after the call to this method.").

Reply:
Done.